olp-cpp-sdk-core/src/client/OlpClientSettings.cpp

```diff
@@ -32,8 +32,21 @@ unsigned int DefaultBackdownPolicy(unsigned int milliseconds) {
 
 bool DefaultRetryCondition(const HttpResponse& response) {
   switch (response.status) {
+    case http::HttpStatusCode::TOO_MANY_REQUESTS:
     case http::HttpStatusCode::INTERNAL_SERVER_ERROR:
+    case http::HttpStatusCode::NOT_IMPLEMENTED:
+    case http::HttpStatusCode::BAD_GATEWAY:
     case http::HttpStatusCode::SERVICE_UNAVAILABLE:
+    case http::HttpStatusCode::GATEWAY_TIMEOUT:
+    case http::HttpStatusCode::VERSION_NOT_SUPPORTED:
+    case http::HttpStatusCode::VARIANT_ALSO_NEGOTIATES:
+    case http::HttpStatusCode::INSUFFICIENT_STORAGE:
+    case http::HttpStatusCode::LOOP_DETECTED:
+    case http::HttpStatusCode::BANDWIDTH_LIMIT_EXCEEDED:
+    case http::HttpStatusCode::NOT_EXTENDED:
+    case http::HttpStatusCode::NETWORK_AUTHENTICATION_REQUIRED:
+    case http::HttpStatusCode::NETWORK_READ_TIMEOUT:
+    case http::HttpStatusCode::NETWORK_CONNECT_TIMEOUT:
```

Comment:
I would change this into:
```cpp
if ((response.status >= INTERNAL_SERVER_ERROR && response.status <= NETWORK_CONNECT_TIMEOUT)
    || response.status == TOO_MANY_REQUESTS) return true;
return false;
```
There are to many status codes between 500 and 599 to list here.